README.md

```diff
@@ -1,2 +1,281 @@
+      postgresql.0.connection_url:       "postgres://postgres:password@database:5432/postgres?sslmode=disable"
+      postgresql.0.max_open_connections: "2"
+      verify_connection:                 "false"
+
+  + vault_database_secret_backend_role.postgres_dev_read
+      id:                                <computed>
+      backend:                           "database"
+      creation_statements:               REDACTED
+      db_name:                           "postgres-secret-backend"
+      default_ttl:                       "2592000"
+      max_ttl:                           "2592000"
+      name:                              "dev-read"
+      revocation_statements:             REDACTED
+
+  + vault_database_secret_backend_role.postgres_service_write
+      id:                                <computed>
+      backend:                           "database"
+      creation_statements:               REDACTED
+      db_name:                           "postgres-secret-backend"
+      default_ttl:                       "864000"
+      max_ttl:                           "864000"
+      name:                              "service-write"
+      revocation_statements:             REDACTED
+
+  + vault_mount.database
+      id:                                <computed>
+      accessor:                          <computed>
+      default_lease_ttl_seconds:         "2592000"
+      max_lease_ttl_seconds:             "2592000"
+      path:                              "database"
+      type:                              "database"
+
+
+Plan: 4 to add, 0 to change, 0 to destroy.
+```
+
+#### Terraform apply
+
+If the `terraform plan` output looks similar to the above one then run `terraform apply` to create the 
+resources.
+
+#### Trying it out
+
+After applying the changes, we should be able to use vault CLI to interact with our newly created database 
+secret backend:
+
+```bash
+# Ensure `VAULT_TOKEN` and `VAULT_ADDR` are set for this work!
+vault read database/config/postgres-secret-backend
+
+Key                                   Value
+---                                   -----
+allowed_roles                         [service-write dev-read]
+connection_details                    map[connection_url:postgres://postgres:*****@database:5432/postgres?sslmode=disable max_open_connections:2]
+plugin_name                           postgresql-database-plugin
+```
+
+##### Use write credentials
+
+First generate a new set of creds with vault:
+
+```bash
+vault read database/creds/service-write
+
+Key                Value
+---                -----
+lease_id           database/creds/service-write/lVpzrysA5akqSvjZVtCgx1i9
+lease_duration     240h
+lease_renewable    true
+password           A1a-9yW06ZdVk54I5KnX
+username           v-token-service--1luYzAYl7SdMxvdpibYv-1555972312
+```
+
+Then use them to login into your local Postgres database and make some changes:
+
+```bash
+psql -h localhost -U v-token-service--1luYzAYl7SdMxvdpibYv-1555972312 -W -d postgres
```

Comment:
Maybe good to assign `username` to a variable or make it clearer/easier to replace that string.